### DISCUSSION TITLE
Timeline for dropping Python 3.6 support

### DISCUSSION THREAD

**Discussion by weiji14** [torchgeo/torchgeo#593] [General]
Continuing on from discussion in https://github.com/microsoft/torchgeo/pull/326 and https://github.com/microsoft/torchgeo/issues/409#issuecomment-1043873350. Python 3.6 has reached end of life (EOL) as of 23 Dec 2021. This issue is to discuss and track when torchgeo should drop Python 3.6 support.

> > Is torchgeo still supporting Python 3.6?
> 
> For the time being, yes. I would prefer to stick to Python's EOL timeline, but there are many systems that still use Python 3.6. I strongly disagree with numpy's aggressive deprecation timeline, as Google Colab still runs Python 3.7 by default. I think we'll likely compromise and stick to whatever versions of Python that PyTorch supports (currently 3.6–3.9 in the latest release, soon to be 3.7–3.9 in the next release). Unless I can convince @calebrob6 to drop 3.6 support since PyTorch has already dropped it in master.
> 

_Originally posted by @adamjstewart in https://github.com/microsoft/torchgeo/issues/409#issuecomment-1043873350_

**Comment by adamjstewart** [comment]
PyTorch 1.11 should be coming pretty soon: https://github.com/pytorch/pytorch/issues/72267

In general, we can always backport things like `dataclasses` and `typing-extensions` if we need to. There are rarely newer language features that we need that can't be backported.

**Comment by calebrob6** [comment]
I only have the slightest preference for not dropping 3.6 (it just seemed silly to do some extra work to make it slightly harder to use the library 😄). PyTorch 1.11 seems like a great reason to drop it.

**Comment by adamjstewart** [comment]
Based on https://pypistats.org/packages/torchgeo very few of our users are using Python 3.6.
